### PR TITLE
Resolves #45: Update path on which call, and env vars and path on install

### LINKED ISSUE #45: which command relies on PATH
The which command relies on the program being findable using PATH, like `where.exe`.

This means it does not work after `fudge install` in a `cmd.exe` , until the `$env:PATH` has been refreshed from the registry, e.g. using https://github.com/chocolatey-archive/chocolatey/blob/master/src/redirects/RefreshEnv.cmd from cmd or https://github.com/chocolatey/choco/blob/stable/src/chocolatey.resources/helpers/functions/Update-SessionEnvironment.ps1 in powershell.

It would be helpful if it also searched for the program using the PATH live in the registry keys, or invoked `Update-SessionEnvironment` before looking for programs.

### PR DESCRIPTION
**Pull request by Badgerati**
Resolves #45

### CODE REVIEW

File: src/Modules/FudgeTools.psm1 [hunk 1]
```diff
@@ -1714,6 +1715,32 @@ function Get-ChocolateyLatestVersion
...
+            if ($items -inotcontains $_) {
+                $items += $_
+            }
+        }
+    }
+
+    $env:PATH = ($items -join ';')
+}
+
+function Update-EnvironmentVariables
```
Comment (jayvdb) [inline src/Modules/FudgeTools.psm1:1761]: I think we need some of tests for this, for specific scenarios.

File: src/Modules/FudgeTools.psm1 [hunk 2]
```diff
@@ -1714,6 +1715,32 @@ function Get-ChocolateyLatestVersion
     return 'latest'
 }
 
+function Update-EnvironmentPath
```
Comment (jayvdb) [inline src/Modules/FudgeTools.psm1:1744]: this looks quite good, and a new creative work for sure.

some new tests would be useful, but not as important as tests for Update-EnvironmentVariables

I think this one can be manually tested.  Ill try to get to it today.

**Comment by jayvdb** [comment]
I recall there were a few variables like `USERNAME` and `PROCESSOR_ARCHITECTURE` which get destroyed when loading them from the registry.  Im not sure if your algorithm will overwrite those.

Also, I recall finding problems with the way choco handled `PSModulePath`.  It at least needs to keep values from the process, but also add values added to the registry which were added by installers, a bit like how your code handles `PATH`.

**Comment by Badgerati** [comment]
I just check the registry values of `USERNAME` and `PROCESSOR_ARCHITECTURE`, versus the process ones. I can confirm that it would destroy the values, so I can make the `Update-EnvironmentVariables` function skip these two keys.

For `PSModulePath`, I was thinking the same thing while reading it. I can make `Update-EnvironmentPath` do both `PATH` and `PSModulePath`.

**Comment by jayvdb** [comment]
Sounds perfect.

**Comment by jayvdb** [comment]
ping @Badgerati ;-)
Would be good to get all these enhancements out into a release.

**Comment by Badgerati** [comment]
Yep! I'll merge these and try and get a release out this week.

Also I started some work on a powershell module, something similar to Fudge but more generic and xplat. If you wanna take a look it's here: https://github.com/Badgerati/Parcel

It's close to being a successor to Fudge, though Fudge will stick around as it's obviously very Chocolatey focused.

**Pull request merged**